helm/cluster-cloud-director/values.yaml

```diff
@@ -9,6 +9,128 @@ cluster:
+          additionalConfig:
+            systemd:
+              units:
+                - contents:
+                    after:
+                      - network-online.target
+                    install:
+                      wantedBy:
+                        - multi-user.target
+                    requires:
+                      - network-online.target
+                    service:
+                      execStart:
+                        - /usr/bin/bash -cv 'echo "$("$(find /usr/bin /usr/share/oem -name vmtoolsd -type f -executable 2>/dev/null | head -n 1)" --cmd "info-get guestinfo.ignition.network")" > /opt/bin/set-networkd-units.sh'
+                        - /usr/bin/bash -cv 'chmod u+x /opt/bin/set-networkd-units.sh'
+                        - /opt/bin/set-networkd-units.sh
+                      remainAfterExit: "yes"
+                      type: oneshot
+                    unit:
+                      description: Install the systemd-networkd interface configuration.
+                  enabled: true
+                  name: set-networkd-units.service
+                - contents:
+                    install:
+                      wantedBy:
+                        - multi-user.target
+                    service:
+                      additionalFields: |-
+                        {{- if $.global.connectivity.network.staticRoutes }}
+                        ExecStart=/usr/bin/bash -cv 'sleep 3'
+                        {{- range $.global.connectivity.network.staticRoutes }}
+                        ExecStart=/usr/bin/bash -cv 'ip route add {{ .destination }} via {{ .via }}'
+                        {{- end }}
+                        {{- else }}
+                        echo "No static routes provided, exiting."
+                        exit 0
+                        {{- end }}
+                      remainAfterExit: "yes"
+                      type: oneshot
+                    unit:
+                      after:
+                        - set-networkd-units.service
+                      description: Add static network routes.
+                      requires:
+                        - set-networkd-units.service
+                  enabled: true
+                  name: static-routes.service
+                - contents:
+                    install:
+                      wantedBy:
+                        - multi-user.target
+                    service:
+                      environment:
+                        - OUTPUT=/run/metadata/coreos
+                      execStart:
+                        - /usr/bin/mkdir --parent /run/metadata
+                        - /usr/bin/bash -cv 'echo "COREOS_CUSTOM_HOSTNAME=$("$(find /usr/bin /usr/share/oem -name vmtoolsd -type f -executable 2>/dev/null | head -n 1)" --cmd "info-get guestinfo.ignition.vmname")" > ${OUTPUT}'
+                        - /usr/bin/bash -cv 'echo "COREOS_CUSTOM_IPV4=$(ip -4 addr show ens192 | awk \'/inet / {print $2}\' | cut -d \'/\' -f1)" >> ${OUTPUT}'
```

Comment:
i don't like hard-coding the interface name into this command, but there should always be an `ens192`.

Reply:
i'll survive with it like this